src/DynamoCore/Graph/Workspaces/PackageDependencyInfo.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dynamo.Graph.Workspaces
+{
+    /// <summary>
+    /// Class containing info about a Dynamo package. 
+    /// Used for serialization.
+    /// </summary>
+    internal class PackageDependencyInfo
+    {
+        /// <summary>
+        /// Name of the package
+        /// </summary>
+        internal string Name { get; set; }
+
+        /// <summary>
+        /// Version of the package
+        /// </summary>
+        internal Version Version { get; set; }
+
+        /// <summary>
+        /// The full name of this package, including its name and version
+        /// Format: name_version
+        /// </summary>
+        internal string FullName { get { return Name + "_" + Version.ToString(); } }
+
+        /// <summary>
+        /// Guids of nodes in the workspace that are dependent on this package
+        /// </summary>
+        internal HashSet<Guid> Nodes
```

Comment:
this seems kind of strange to me... how do we know which workspace these nodes refer to? Can you only retrieve or generate these objects from a workspace?